packages/sirv/index.js

```diff
@@ -69,59 +83,84 @@ function send(req, res, file, stats, headers={}) {
 	let setHeaders = opts.setHeaders || noop;
 
-	if (opts.dev) {
-		return function (req, res, next) {
-			let stats, file, uri=decodeURIComponent(req.path || req.pathname || parser(req).pathname);
-			let arr = [uri].concat(toAssume(uri, extensions)).map(x => normalize(join(dir, x))).filter(x => {
-				return x.startsWith(dir) && fs.existsSync(x);
-			});
-
-			while (file = arr.shift()) {
-				stats = fs.statSync(file);
-				if (stats.isDirectory()) continue;
-				setHeaders(res, uri, stats);
-				return send(req, res, file, stats, {
-					'Content-Type': mime.getType(file),
-					'Last-Modified': stats.mtime.toUTCString(),
-					'Content-Length': stats.size,
-				});
-			}
-			return next ? next() : isNotFound(req, res);
-		}
+	let extensions = opts.extensions || ['html', 'htm'];
+	let gzips = opts.gzip && extensions.map(x => `${x}.gz`).concat('gz');
+	let brots = opts.brotli && extensions.map(x => `${x}.br`).concat('br');
+
+	let fallback = '/';
+	let isEtag = !!opts.etag;
+	let isSPA = !!opts.single;
+	if (typeof opts.single === 'string') {
+		let idx = opts.single.lastIndexOf('.');
+		fallback += !!~idx ? opts.single.substring(0, idx) : opts.single;
+	}
+
+	let ignores = [];
+	if (opts.ignores !== false) {
+		ignores.push(/\w\.\w+$/); // any extn
+		if (opts.dotfiles) ignores.push(/\/\.\w/);
+		else ignores.push(/\/\.well-known/);
+		[].concat(opts.ignores || []).forEach(x => {
+			ignores.push(new RegExp(x, 'i'));
+		});
 	}
 
 	let cc = opts.maxAge != null && `public,max-age=${opts.maxAge}`;
 	if (cc && opts.immutable) cc += ',immutable';
 
-	list(dir, (name, abs, stats) => {
-		if (!opts.dotfiles && name.charAt(0) === '.') {
-			return;
-		}
+	if (!opts.dev) {
+		list(dir, (name, abs, stats) => {
+			if (/\.well-known[\\+\/]/.test(name)) {} // keep
+			else if (!opts.dotfiles && /(^\.|[\\+|\/+]\.)/.test(name)) return;
 
-		let headers = {
-			'Content-Length': stats.size,
-			'Content-Type': mime.getType(name),
-			'Last-Modified': stats.mtime.toUTCString(),
-		};
+			let headers = toHeaders(name, stats, isEtag);
+			if (cc) headers['Cache-Control'] = cc;
 
-		if (cc) headers['Cache-Control'] = cc;
-		if (opts.etag) headers['ETag'] = `W/"${stats.size}-${stats.mtime.getTime()}"`;
+			FILES['/' + name.normalize().replace(/\\+/g, '/')] = { abs, stats, headers };
+		});
+	}
 
-		FILES['/' + name.replace(/\\+/g, '/')] = { abs, stats, headers };
-	});
+	let lookup = opts.dev ? viaLocal.bind(0, dir, isEtag) : viaCache;
 
 	return function (req, res, next) {
-		let pathname = decodeURIComponent(req.path || req.pathname || parser(req).pathname);
-		let data = FILES[pathname] || find(pathname, extensions);
+		let extns = [''];
+		let val = req.headers['accept-encoding'] || '';
+		if (gzips && val.includes('gzip')) extns.unshift(...gzips);
+		if (brots && /(br|brotli)/i.test(val)) extns.unshift(...brots);
+		extns.push(...extensions); // [...br, ...gz, orig, ...exts]
+
+		let pathname = req.path || parser(req, true).pathname;
```

Comment:
Rollback please decodeURIComponent because this issue is back: https://github.com/lukeed/sirv/pull/21